### ISSUE TITLE
t.im.basechat.ChatUI.contactChangedNick broken

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7501]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#229|
|Type|enhancement|
|Created|2003-08-24 01:41:33Z|
|Branch|https://github.com/twisted/twisted/tree/contact-changed-nick-229|

<details><summary>Searchable metadata</summary>

```
trac-id__229 229
type__enhancement enhancement
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__branches_contact_changed_nick_229 branches/contact-changed-nick-229
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__words words
keywords__im im
time__1061689293000000 1061689293000000
changetime__1268515031000000 1268515031000000
version__ 
owner__ 
cc__exarkun cc__acapnotic cc__fedora cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
It references a variable, "person" which does not exist.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="keturn's avatar" src="https://avatars.githubusercontent.com/u/83819?s=50" width="50" height="50">](https://github.com/keturn)|@keturn commented|
|-|-|

```
#!html
<pre>
See my commit message for revision 1.8 (2/11/2003) -- that's
probably when I broke this.  What would you like to do about it?

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
im is deprecated and documented as not-really-working
Therefore, downgrading to feature.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="keturn's avatar" src="https://avatars.githubusercontent.com/u/83819?s=50" width="50" height="50">](https://github.com/keturn)<a name="note_4"></a>|@keturn commented|
|-|-|

google code search says nobody uses basechat.ChatUI.  Can we delete it?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

The minchat example (one of the most run examples in Twisted, I think) uses it.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess commented|
|-|-|

Is there any way for me to see what revision 1.8 changed?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Looks like it corresponds to r5411

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess commented|
|-|-|

(In [28628]) Branching to 'contact-changed-nick-229'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess commented|
|-|-|

(In [28632]) Use variables that actually exit and add unit tests for basechat.ChatUI.

refs [#7501](https://github.com/twisted/twisted/issues/7501)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_12"></a>|@glyph set owner to @jesstess|
|-|-|

A few minor issues:

  1. There's some trailing whitespace in the tests.
  1. The test module should have a module docstring (it should be pretty trivial, just point at the `basechat` module)
  1. The tests refer to a 'person', they should probably have epytext links to `IPerson`.
  1. `ChatUI`'s docstring isn't really accurate; it's not a GUI, it's sort of an event hub.  Not really a problem for this branch, just thought I'd mention it.  (I can't immediately think of a better thing though.)

Otherwise this really looks pretty good as is; address these issues to your satisfaction and then merge.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_13"></a>|@jesstess set status to `closed`|
|-|-|

Thanks for the review glyph. Your feedback was addressed in the branch, and then:

Merge contact-changed-nick-229

Author: jesstess
Reviewer: glyph
Fixes: [#7501](https://github.com/twisted/twisted/issues/7501)

twisted.words.im.basechat.ChatUI now has a functional
contactChangedNick with unit tests.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
